src/ArborX_HDBSCAN.hpp

```diff
@@ -0,0 +1,54 @@
+/****************************************************************************
+ * Copyright (c) 2017-2022 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+#ifndef ARBORX_HDBSCAN_HPP
+#define ARBORX_HDBSCAN_HPP
+
+#include <ArborX_Dendrogram.hpp>
+#include <ArborX_MinimumSpanningTree.hpp>
+
+#include <Kokkos_Profiling_ProfileSection.hpp>
+
+#include <iomanip>
+#include <limits>
+
+namespace ArborX::Experimental
+{
+
+template <typename ExecutionSpace, typename Primitives>
+auto hdbscan(ExecutionSpace const &exec_space, Primitives const &primitives,
+             int core_min_size)
+{
+  Kokkos::Profiling::pushRegion("ArborX::HDBSCAN");
+
+  using MemorySpace = typename Primitives::memory_space;
+
+  Kokkos::Profiling::ProfilingSection profile_mst("ArborX::HDBSCAN::mst");
+  profile_mst.start();
```

Comment:
Why are the regions `"ArborX::HDBSCAN"` with the nested `"ArborX::MST::MST"` and `"ArborX::Dendrogram::Dendrogram"` not sufficient?

Reply:
These are profiling sections that we'll use to print out the timers. When using Kokkos tooling like `space-time-stack`, those would have been sufficient.